### DISCUSSION TITLE
How to configure for Unity WebGL

### DISCUSSION THREAD

**Discussion by boylie3d** [vercel/vercel#4954] [Help] [answered]
We have site deploying Unity WebGL content, and we're trying to speed up loading by enabling Brotli compression and WebAssembly streaming. In order to do so, Unity provides examples on how to configure an Apache or IIS server correctly. [That documentation is here](https://docs.unity3d.com/Manual/Webgl-Server-configuration-for-WebAssembly-streaming.html). I'm relatively new to Vercel, and I believe my answer here is to configure a vercel.json file to accomplish this, I'm just a little unsure on how to do it, based on the examples in the docs. If anyone could provide some clarity, it'd be much appreciated

**Comment by styfle** [comment, marked as answer]
Hi @boylie3d 

Vercel is already configured to serve [Brotli Compression](https://vercel.com/docs/v2/edge-network/compression) when the client supports it.

Additionally, `.wasm` files are already served with the `application/wasm` content-type header.

No configuration needed 👍

**Reply by omaramen99** [reply]
Hi from 2023, I don't know if this got changed or not but I think I have an issue with unity webgl brotli build (and also the gzip)
![br](https://user-images.githubusercontent.com/56793975/212052985-e3616ebe-d9f5-4336-bbc6-9dc72a512646.png)

**Reply by boylie3d** [reply]
Granted this was a WHILE ago, but I went and found an old project we had online - the fix was in setting up a `vercel.json` file at the root of the project - you can find more on that [here](https://vercel.com/docs/project-configuration)

We needed to set some headers in that file, this is what we had that eventually, I THINK (like 90% confident) worked for our  builds

```
  "headers": [
    {
      "source": "/(.*\\.unityweb)",
      "headers" : [
        {
          "key" : "Content-Encoding",
          "value" : "gzip"
        }
      ]
    },
    {
      "source": "/(.*\\.wasm\\.unityweb)",
      "headers" : [
        {
          "key" : "Content-Type",
          "value" : "application/wasm"
        }
      ]
    },
    {
      "source": "/(.*\\.gz)",
      "headers" : [
        {
          "key" : "Content-Encoding",
          "value" : "gzip"
        }
      ]
    },
    {
      "source": "/(.*\\.wasm\\.gz)",
      "headers" : [
        {
          "key" : "Content-Type",
          "value" : "application/wasm"
        }
      ]
    },
    {
      "source": "/(.*\\.br)",
      "headers" : [
        {
          "key" : "Content-Encoding",
          "value" : "br"
        }
      ]
    },
    {
      "source": "/(.*\\.wasm\\.br)",
      "headers" : [
        {
          "key" : "Content-Type",
          "value" : "application/wasm"
        }
      ]
    }
  ]
```

Hope this helps!

**Reply by shohamjac** [reply]
Thank you, fixed it for me as well!
